### ISSUE TITLE
Multiple learning rates support for LinearWarmupCosineAnnealingLR

### ISSUE THREAD

**Issue by Magolor** [PyTorchLightning/lightning-bolts#764]
## 🚀 Feature

Multiple learning rates support for LinearWarmupCosineAnnealingLR

### Motivation

Sometimes when training a model, different learning rates are used for different parts: for example, tuning a model with pre-trained backbone, the classifier parameters' learning rates should be larger than those of the backbone parameters.

However, this can not be directly done in the pytorch-lightning framework using the multi-optimizer grammar, since multi-optimizer behavior in pytorch-lightning is GAN-like, which runs the network for each optimizer separately. Though, this can be done using `automatic_optimization=False`, it is not elegant and it would weaken the modularity since I now have to consider manually gradient clipping, gradient accumulation, etc.

### Pitch

If scheduler is not used, pytorch optimizers naturally supports parameter groups, which could solve this problem. Thus it would be nice if scheduler also supports parameter groups, connected by a common base lr and ratio. That is, for example, if I set the backbone learning rate to be 10x classifier learning rate, I expect the warmup learning rate and the final epoch learning rate to be proportionally set for different parameter groups in the optimizer, too.

**Comment by ananyahjha93** [comment]
We are deprecating the self-supervised learning paper implementations in bolts and instead provided a [VISSL](https://vissl.ai/) integration in PyTorch Lightning Flash which can be used to train models using self-supervised learning algorithms. You can refer to the documentation [here](https://lightning-flash.readthedocs.io/en/latest/reference/image_embedder.html).
